### ISSUE TITLE
Encountered fatal Python error with TFLite Experimental Converter

### ISSUE THREAD

**Issue by willbattel** [tensorflow/tensorflow#40233]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): `tensorflow/tensorflow:latest-gpu` Docker image on Ubuntu 20.04 host on GCP Compute Engine VM
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v2.2.0-rc4-8-g2b96f3662b 2.2.0
- Python version: 3.6.9
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: 10.2 (Using Docker TensorFlow GPU image)
- GPU model and memory: Tesla V100-SXM2-16GB

**Describe the current behavior**
I'm using the new QAT API demonstrated in the docs [here](https://www.tensorflow.org/model_optimization/guide/quantization/training_example). Then I'm taking the resulting model and converting it to TFLite, again referencing the docs. Everything works as advertised until converting the model, where I encounter an error (see logs below).

**Describe the expected behavior**
I expected to get a converted model I could save as a `tflite` binary. I was only able to do this when explicitly disabling the experimental converter.

**Standalone code to reproduce the issue**
I don't have this available at the moment- I will try to provide this when able. In the meantime, this is the code snippet surrounding the problem. Let me know if this is insufficient.

```
model = tfmot.quantization.keras.quantize_model(model)
model.compile(optimizer=optimizer, loss=loss)
history = model.fit(train_dataset, epochs=FLAGS.epochs, callbacks=callbacks, validation_data=val_dataset)

converter = tf.lite.TFLiteConverter.from_keras_model(model)
# converter.experimental_new_converter = False # Must uncomment this line to convert successfully
converter.optimizations = [tf.lite.Optimize.DEFAULT]
tflite_model = converter.convert()

with open("output.tflite", 'wb') as f:
    f.write(tflite_model)
```

Our model is made up of the following Keras layers: Add, Concatenate, Conv2D, Input, Lambda, LeakyReLU, MaxPool2D, UpSampling2D, ZeroPadding2D, and BatchNormalization organized into several sub-models. Each submodel is quantized on its own because the `tfmot` API does not have native support for submodels. [This ](https://github.com/tensorflow/model-optimization/issues/377#issuecomment-623586866) was the recommended workaround until that is added.

**Other info / logs**
It's a long one, here is what was spit out. All lines following line 59 were output together at the end. From my point of view the console didn't update from line 59 for several minutes until it all appeared together. I checked `nvidia-smi` while I waited and noticed GPU VRAM usage was at 100% (almost 16GB was being used by the Python process running this script) but GPU utilization was at 0%. Checking `htop` showed RAM and CPU usage very low at the time.

https://pastebin.com/LMNefb3V

**Comment by amahendrakar** [comment]
@willbattel,
In order to expedite the trouble-shooting process, could you please provide the complete code or the `model` file you are using in the code. Thanks!

**Comment by willbattel** [comment]
Sure, I'll try to bring that code over. I need to clean it up first.

**Comment by amahendrakar** [comment]
> Sure, I'll try to bring that code over. I need to clean it up first.

@willbattel,
Any updates regarding this? Thanks!

**Comment by willbattel** [comment]
> Any updates regarding this? Thanks!

Not yet. I've been delayed by other work that is higher priority because this issue wasn't blocking for us. Because it works if you disable the experimental converter, I just wanted to submit the issue for your reference. If I find time to submit a reproducible sample, I'll add that in this thread- I can't just drop in what we ran because it is littered with clutter, proprietary code, etc.... In the mean time, if you would like to close this issue you can do so. We could re-open it later, or I could always submit a new issue if need-be.

**Comment by amahendrakar** [comment]
@willbattel,
Thank you for the update. Marking the issue as close, please feel free to reopen if necessary. Thanks!
